src/MarkersHandler.js

```diff
@@ -316,12 +317,23 @@ var markersHandler = function (vjs, adMarkerStyle) {
 	          return setting.markerTip.time(markersList[index + 1]);
 	        }
 	        // next marker time of last marker would be end of video time
-	        return player.duration();
+	        return _videoDuration;
 	      };
 	      var currentTime = player.currentTime();
 	      var newMarkerIndex = NULL_INDEX;
 
-	      var nextMarkerTime;
+				var nextMarkerTime;
+
+				// post-roll support
+				if (currentTime === player.duration()) {
```

Comment:
1. I did try with few videos with different durations and I see the Brightcove player (not video tag) always fires 'timeupdate' event exactly at the end of the video. In this case post-roll works most nicely.
2. I'm specially checking the real player duration for post-roll because the real duration could be litle different the duration we recieved after 'loadedmetadata' event fired.